internal/cache/pruner.go

```diff
@@ -60,7 +60,7 @@ func (c *AutoPruneConfig) pruneTimer(background context.Context, pruner Pruner,
 		case <-timer.Chan():
 			timer.Reset(c.Interval)
```

Comment:
Strictly speaking, doing the reset after the job would break the defined interval, right?
For example if the job is configured to run every 10 minutes, and the first job runs 12:00, it is logic to expect the next job to run at 12:10, 12:20, 12:30 etc.

If we reset the timer after the job has ran, one would experience a drift equal to the job duration. So if the job takes 1 minute, the jobs will run at 12:00, 12:11, 12:22.

If that's not important, I might as well change it. It might as well be an expectation in my mind coming from my past with automation. There timers typically must not drift.